src/Temporalio/Bridge/Scope.cs

```diff
@@ -53,21 +60,65 @@ public Interop.TemporalCoreByteArrayRef ByteArray(string? str)
         }
 
         /// <summary>
-        /// Create a metadata byte array ref.
+        /// Create a key-value pair byte array ref.
         /// </summary>
-        /// <param name="metadata">Metadata to create from.</param>
+        /// <param name="pair">Key-value pair to create from.</param>
         /// <returns>Created byte array ref.</returns>
-        public Interop.TemporalCoreByteArrayRef Metadata(IEnumerable<KeyValuePair<string, string>>? metadata)
+        public Interop.TemporalCoreByteArrayRef ByteArray(KeyValuePair<string, string> pair)
         {
-            if (metadata == null)
+            if (pair.Key.Length == 0)
             {
                 return ByteArrayRef.Empty.Ref;
             }
-            var val = ByteArrayRef.FromMetadata(metadata);
+            var val = ByteArrayRef.FromKeyValuePair(pair);
             byteArrayRefs.Add(val);
             return val.Ref;
         }
 
+        /// <summary>
+        /// Create a key-value pair byte array ref.
+        /// </summary>
+        /// <param name="pair">Key-value pair to create from.</param>
+        /// <returns>Created byte array ref.</returns>
+        public Interop.TemporalCoreByteArrayRef ByteArray(KeyValuePair<string, byte[]> pair)
+        {
+            if (pair.Key.Length == 0)
```

Comment:
This conditional is a bit interesting. Is it just in case, or are there situations where we think it may happen? Same for the conditional in the other overload of this.